src/Components/ADT3DSceneBuilder/Internal/VisualRuleForm/VisualRuleForm.tsx

```diff
@@ -60,89 +74,169 @@ const VisualRuleForm: React.FC<IVisualRuleFormProps> = (props) => {
         theme: theme
     });
     const commonFormStyles = getPanelFormStyles(theme, rootHeight);
-    const typeOptions: Array<IDropdownOption> = useMemo(
-        () =>
-            Object.keys(DTDLPropertyIconographyMap).map((mappingKey) => ({
-                key: `value-type-${DTDLPropertyIconographyMap[mappingKey].text}`,
-                text: DTDLPropertyIconographyMap[mappingKey].text,
-                data: { icon: DTDLPropertyIconographyMap[mappingKey].icon }
-            })),
-        []
+
+    // Refs
+    // This combination of init function and useRef should replace a useEffect that runs onMount
```

Comment:
I can remove this comment, but the intent of it was to explain why I used a ref here. I think we rely too much on using useEffect to initialize values, I am trying this pattern to avoid the 'onMount' useEffects we have in other places.

Reply:
From: https://overreacted.io/a-complete-guide-to-useeffect/, seems like refs are favored over useEffect as onMount value set.

![image](https://user-images.githubusercontent.com/55852250/195670868-0389c7d1-ee76-4262-a953-1c377d20c04b.png)